### DISCUSSION TITLE
Z-pinning to increase isotropy of part strength

### DISCUSSION THREAD

**Discussion by INDIVISUM** [OrcaSlicer/OrcaSlicer#4815] [Ideas]
The common problem 3d printed parts are that their tensile strength is anisotropic. i.e. parts are weaker in the Z direction than in the X/Y directions. Z-pinning increases the strength and toughness if parts in the z direction compared to not using Z-pinning. to quote the research paper (linked below): This initial investigation of z-pin printed structures with unreinforced PLA showed 20% increase in strength and 100% increase in toughness in the z-direction. The fracture surfaces of z-pinned samples were very rough, indicating that the pins successfully re-routed the crack front as it propagated between layers

As described in the research paper:
“The z-pinning process begins by printing a part in a conventional fashion that contains multiple voids in each layer that are intentionally aligned in the z-axis (Fig 1a). In the example shown, the voids are staggered in depth – one penetrating 6 layers deep, the other only 3 layers. Prior to deposition of the next layer, the print head is positioned over the deeper of the two holes and material is extruded until the void is filled (Fig 1b). The shallower of the two holes remains unfilled. Then the next 3 layers are conventionally printed, maintaining the aligned holes across layers (Fig 1c). The print head is then positioned over the deeper of the holes (different x-y location) and material is extruded in the z-direction to fill the void (Fig 1d). The process is repeated for the following 3 layers, allowing for deposition of a second pin into the same hole (x-y location) as the original pin (Fig 1e). Although this approach allows for deposition of continuous material across multiple layers, the z-pins have a finite length (6 layers in this example) which results in a “seam” at the interface between successive pins at a given location. The seam position can be varied among neighboring pins by staggering either the pin length or the starting layer of the pin.”

![Z-pinning fig 1](https://github.com/SoftFever/OrcaSlicer/assets/34963308/0124e1cf-01b6-48ac-a430-3c498e1e436c)

![Z-pinning fig 2](https://github.com/SoftFever/OrcaSlicer/assets/34963308/c95383a5-e298-4031-8f06-8f793c51757d)

![Z-pinning fig 3](https://github.com/SoftFever/OrcaSlicer/assets/34963308/d2ce92e5-d92a-4ad1-84f9-116be174f6da)

Please refer to this research paper: 
https://www.osti.gov/biblio/1808415

**Comment by shakezilla19** [comment]
Brilliant idea!

Wish I knew how to code because I'd love to make this happen. Maybe final implementation could be an infill setting something like "Z-Pin Rectilinear". 

The big problem appears to be how to tell the slicer where to add the pins or to make it automatic. I imagine the research group just sliced their model then hacked the gcode to have an action similar to an initial purge line which is pretty easy but very time consuming. A semi automatic approach seems to be the next step. Something like a "paint Z pin" feature in the slicer. The paint locations would then add a "purge line action" to the areas after every 5 layers, or so, and there is your Z-pin. 

A basic Gcode to me would look like:
Prior Gcode.... 
G1 X, Y ; Move to Pin1
G1 E1 ;Extruder creates Z-pin,
G1 X, Y ; Move to Pin2
G1 E1 ;Extruder creates Z-pin
ect....

**Comment by foresterr** [comment]
Wow, awesome. That would be great for functional parts. You probably wouldn't want to have entire model using this (it would be practically 100% infill), but an additional "shell" of this structure right under the walls would strenghten the model without increasing use of the filament dramatically.

In layman terms, it's leaving small vertical holes in the model, and then pumping molten filament straight into them, right? I guess getting the amount and pin length right for all cases might be tricky.

**Comment by vgdh** [comment]
Hmm. But where the internal air would go when the upper of the bore sealed with the filament during z-pinning process.
It's need some experimental testing to check if that possible.

**Reply by TMRolle** [reply]
Maybe you could use U-shaped holes? Or just a small horizontal vent? May be a non issue too, most prints are not fully airtight, plus air is compressible, plus nozzles can build up a fairly large amount of pressure, on the order of a few hundred PSI according to this paper: https://pubs.aip.org/sor/jor/article/63/1/141/848523/In-line-rheological-monitoring-of-fused-deposition#_i27

**Reply by pvt02** [reply]
or slots/canyons/channels

**Reply by CSV-12** [reply]
The air? Turn on the vacuum pump!

**Comment by Anthony-Bec** [comment]
+1

**Comment by gundy360** [comment]
It would be interesting if you could have the z pinning right next to the shell and tie it into the rest of the part in an interlocking way. To more effectively tie into the rest of the part for the layers leading up to the pinning, you could do the regular infill with a small additional circle around the pinning site being printed amongst the infill, but not breaking it. That way when the pinning actually happens, the plastic interlocks around the sparse filaments in the pinning region further tying the layers together. The circular paths amongst the previous layers would just act to contain the z pinning plastic. If venting could be achieved by leaving a small space on the first layer of the feature to allow excess to escape into the infill region on the bottom.

**Comment by KestutisTauckela** [comment]
Thank you for sharing your idea, especially with images and such a detailed explanation. Impressive!
I had very similar idea with only small "enhancement": shape is a **vertical dumbbell** instead of a line.

Let's say for standard 0.4mm nozzle we leave 1.2mm disc of emptiness in first layer, 0.4mm disc in second layer, 0.4mm disk in third layer and again 1.2mm disc of emptiness in fourth layer. Then move nozzle directly above the empty dumbbell shape and fill it all with plastic in one go. In my theory, that should create a sort of **rivet style solid structure**, binding two middle layers firmly and somewhat reinforcing first and fourth layers. Whole structure does not need to be exactly 4 layers. It could be more (perhaps configurable via parameters). Not sure about 3 layer structures... Needs to be tested.

I really hope someone will find time to implement this. I'd like this idea to be free and benefit every Orca Slicer user.

**Comment by KestutisTauckela** [comment]
Here is a diagram, visualizing my description. Hope this helps:
![Z layers reinforcement](https://github.com/user-attachments/assets/e6a0234f-c51a-4379-9d4c-23e7c4360fc3)

**Reply by pvt02** [reply]
If this was combined with my slot/canyon/channel idea; your idea would look like embedded I-beams. I mean it looks like you designed little I-beams already. I was unclear how long you thought of these being. And maybe?? the tops and bottoms of the I-beams could spread wider to overlap more and create a stronger bond/lock? This would be particularly neat if the user decided to infill a non carbon fiber material with carbon fiber I-beams. The I-beams would have fibers going in more than just the single linear directions as they do now, maybe.

**Comment by pvt02** [comment]
Hello. I am no coder, but I realize that this is a great idea in general!

BUT if instead of holes for Z-pinning, what about Z-canyoning/Z-slotting? In other words, leave areas of canyons/slots/channels to be filled later. This might do two things: 
1. have more surface area contact across horizontal layers for more Z tensile strength 
2. much reduced hot-end stopping and starting for many holes. Also I read the discussion above about venting for pinning holes, and the slot/canyon idea simply vents as the plastic pushes to the bottom, displacing the air ahead of the plastic and up into the chamber.

Beyond this basic idea:
A. this could be combined in the brick layers option 
B. the slots can of course be offset initiated as noted for "pinning" in the discussion (staggered); but an adjustable gap option above/below canyons may be desirable, so that if you are pinning with a second material (say carbon fiber in a non carbon fiber material) the tougher non-carbon fiber material can Z-weave (needs anther name??) in the horizontal plane under and over canyons so there is not a single canyon top to bottom through the entire print, possible causing a new issue,  "vertical layer line" weak points (z horizontal strength problem transferred to the vertical) 
C. slot depth can be adjustable 2-10+ layers deep?? after experiments, this can default to a pretty good value, while still having depth options. 
D. the slots can criss-cross if desired. 
E. does this need its own flow rate adjustment option?? 
F. Instead of simply canyons/slots in an otherwise horizontal layered part, perhaps something more encompassing can be done, like the whole part canyoned/slotted with an equal amount of horizontal "gapped" layers, so that the overall effect is layers 50%/50% going both ways and perhaps coming close to true isotropic strength?? 
G. These micro-canyons/slots/channels might also be implemented in the walls specifically (separate options for walls and infill??) to basically have the curve/shape of the walls, being an option of how the walls are printed.
H. Would it be helpful to have an option to increase the slot extrusion material temperature to weld everything even better? It is not going to sag, since it is simply settling in a void, right? I mean up to a point, as long as it is not so extreme it deforms the part.

**Reply by arhi** [reply]
how about you just model that into the part and then configure that part as 100% infill compared to rest of the part?
you can't really deterministically fill complex shapes, even the pinning thing is very limited.

**Reply by pvt02** [reply]
Since Z strength is the main limiting factor for part strength; this seems to be a worthwhile investigation for a slicer feature. I also don't see a slot as complex. It seems to me that history is filled with those that say something can not be done, just to have that "impossibility" overcome later.

**Reply by arhi** [reply]
> It seems to me that history is filled with those that say something can not be done

Feel free to work on it :) we will all appreciate it.

Problem is that you cannot leave the "channel" open and then position head somewhere and fill it as you do not have solid walls and you are not pouring liquid but the molten plastic that you are pouring melts those walls and does not flow nicely. You could then fill in the channel 3d printing it with controlled moves like you are doing with the rest of the print so not much point really.
As for slicer figuring out beams and channels, I would really appreciate slicer still being CAM and not CAD software and keep doing CAD stuff in CAD not to waste valuable time on CAM development adding it CAD features, especially ones already available in CAD. But as I said, if you have some spare time, noone will have anything against you giving it a try.

**Comment by MGunlogson** [comment]
I've added experimental features to do something similar to Z-pinning. A new infill type that makes U shaped vertical channels and injects plastic into one side at the top. I'm calling it MAGMA because its almost like a lava tube, but beneath the surface of part

I have binaries ready if anyone wants to play with it. I haven't got a clean print yet, maybe those with better hardware than my Ender 3 will have better luck.

New infill type, dozens of settings, "dual infill shell" feature. Etc

Check it out!

Project overview
https://mgunlogson.github.io/magma/
PR
https://github.com/OrcaSlicer/OrcaSlicer/pull/14122
Config reference
https://github.com/MGunlogson/magma/blob/main/RELEASE.md
